### PR TITLE
[TIMOB-25269] (6_3_X) Support iPhone X launch-screens

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25269

Backport of https://github.com/appcelerator/titanium_mobile/pull/9423

### CODE REVIEW

File: iphone/Classes/TiRootViewController.m [hunk 1]
```diff
@@ -316,42 +316,56 @@ - (UIImage*)defaultImageForOrientation:(UIDeviceOrientation) orientation resulti
...
-    if ([TiUtils isRetinaHDDisplay]) {
+  
+    // iPhone X
+    if ([TiUtils isRetinaiPhoneX]) {
+      if (UIDeviceOrientationIsPortrait(orientation)) {
+        // Portrait
+        image = [UIImage imageNamed:@"LaunchImage-1100-Portrait-2436h@3x"];
+      } else if (UIDeviceOrientationIsLandscape(orientation)) {
+        // Landscape
+        image = [UIImage imageNamed:@"LaunchImage-1100-2436h@3x"];
```
Comment (rborn) [inline iphone/Classes/TiRootViewController.m]: Hi @hansemannn , any reason for the image not to have `landscape` in name? Thnx!

Comment (hansemannn) [reply iphone/Classes/TiRootViewController.m]: Indeed. Xcode seems to store it as `LaunchImage-1100-2436@3x.png` inside the bundle, see this image from my [post at StackOverflow](https://stackoverflow.com/questions/46263795/iphone-x-incorrect-launch-screen-orientation-used):

Comment (hansemannn) [reply iphone/Classes/TiRootViewController.m]: Which is incorrect, but it seems the way Xcode does it. I opened a radar already and awaiting feedback from Apple today.

**Review by cb1kenobi — APPROVED** [review]
Looks good! **APPROVED**

**Pull request merged**